checkov/terraform/checks/resource/azure/AKSMaxPodsMinimum.py

```diff
@@ -18,16 +18,26 @@ def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
         max_pods = 30  # default is 30
 
         self.evaluated_keys = ["max_pods"]
-        if "max_pods" in conf.keys() and isinstance(conf["max_pods"][0], int):
-            max_pods = conf["max_pods"][0]
-
-        if "default_node_pool" in conf.keys():
+        pods = conf.get("max_pods")
+        if pods and isinstance(pods, list):
+            pods = pods[0]
+            if self._is_variable_dependant(pods):
```

Comment:
Isn't `if not isinstance(pods, int)` enough here?
Aren't all the checks we're doing in _is_variable_dependant excessive for this case? :)